expand-contract-upgrade/cluster_ops.sh

```diff
@@ -138,16 +162,60 @@ create_cluster() {
 }
 
 
-## setup the example application
-setup_app() {
-  echo ""
-  echo "Setting up the application ....."
+## Creates the Shakespeare index and loads the data
+load_data() {
+  echo "Setting up port-forward to Elasticsearch client"
+  kubectl port-forward svc/elasticsearch 9200 1>&2>/dev/null &
+  # Wait a couple seconds for connection to establish as that last command is
+  # not blocking
+  sleep 5
+
+  echo "Creating the Shakespeare index"
+  # The mapping file creates the index and sets the metadata needed by
+  # Elasticsearch to parse the actual data
+  curl -H "Content-Type: application/json" \
+    -X PUT \
+    -d @"${REPO_HOME}/data/mapping.json" \
+    'http://localhost:9200/shakespeare'
+  # The response does not make include a newline
   echo ""
-  kubectl create -f "${REPO_HOME}/manifests/hello-server.yaml"
-  kubectl create -f "${REPO_HOME}/manifests/hello-svc.yaml"
+
+  # Here we load the actual data.
+  echo "Loading Shakespeare sample data into Elasticsearch"
+  curl -H "Content-Type: application/x-ndjson" \
+    -X POST \
+    --data-binary @"${REPO_HOME}/data/shakespeare.json" \
+    'http://localhost:9200/shakespeare/doc/_bulk?pretty' > /dev/null
+
+  # If we've made it this far the data is loaded
+  echo "Sample data successfully loaded!"
+
+  pkill -P $$
```

Comment:
Is killing the script going to kill the kubectl in the background?

Reply:
Yes, `-P` searches for processes with a parent pid matching the given number and `$$` is the PID of this executing script.  So, this command will kill anything that was spawned by this script.